test/unit/org/apache/cassandra/utils/AbstractTypeGenerators.java

```diff
@@ -72,42 +103,271 @@
-              // LexicalUUIDType.instance,
-              // SimpleDateType.instance,
-              // TimeType.instance,
-              // DurationType.instance,
+              TypeSupport.of(EmptyType.instance, rnd -> null, (a, b) -> 0),
+              TypeSupport.of(DurationType.instance, CassandraGenerators.duration(), Comparator.comparingInt(Duration::getMonths)
+                                                                                              .thenComparingInt(Duration::getDays)
+                                                                                              .thenComparingLong(Duration::getNanoseconds))
     ).collect(Collectors.toMap(t -> t.type, t -> t));
     // NOTE not supporting reversed as CQL doesn't allow nested reversed types
     // when generating part of the clustering key, it would be good to allow reversed types as the top level
-    private static final Gen<AbstractType<?>> PRIMITIVE_TYPE_GEN = SourceDSL.arbitrary().pick(new ArrayList<>(PRIMITIVE_TYPE_DATA_GENS.keySet()));
+    private static final Gen<AbstractType<?>> PRIMITIVE_TYPE_GEN;
+    static
+    {
+        ArrayList<AbstractType<?>> types = new ArrayList<>(PRIMITIVE_TYPE_DATA_GENS.keySet());
+        Collections.sort(types, Comparator.comparing(a -> a.getClass().getName()));
+        PRIMITIVE_TYPE_GEN = SourceDSL.arbitrary().pick(types);
+    }
+    private static final Set<Class<? extends AbstractType>> NON_PRIMITIVE_TYPES = ImmutableSet.<Class<? extends AbstractType>>builder()
+                                                                                              .add(SetType.class)
+                                                                                              .add(ListType.class)
+                                                                                              .add(MapType.class)
+                                                                                              .add(TupleType.class)
+                                                                                              .add(UserType.class)
+                                                                                              .add(VectorType.class)
+                                                                                              .build();
 
     private AbstractTypeGenerators()
     {
 
     }
 
     public enum TypeKind
-    {PRIMITIVE, SET, LIST, MAP, TUPLE, UDT}
+    {PRIMITIVE, SET, LIST, MAP, TUPLE, UDT, VECTOR}
 
     private static final Gen<TypeKind> TYPE_KIND_GEN = SourceDSL.arbitrary().enumValuesWithNoOrder(TypeKind.class);
 
+    public static Set<Class<? extends AbstractType>> knownTypes()
+    {
+        Set<Class<? extends AbstractType>> types = PRIMITIVE_TYPE_DATA_GENS.keySet().stream().map(a -> a.getClass()).collect(Collectors.toSet());
+        types.addAll(NON_PRIMITIVE_TYPES);
+        types.add(FrozenType.class);
+        types.add(ReversedType.class);
```

Comment:
not tested here but is tested in `CasssandraGenerators.TableMetadataBuilder`... this is here so the test that makes sure all cases are covered doesn't fail for this type